lib/Models/GenericParameter.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+/*global require*/
+var defineProperties = require('terriajs-cesium/Source/Core/defineProperties');
+var FunctionParameter = require('./FunctionParameter');
+var inherit = require('../Core/inherit');
+
+/**
+ * A parameter that can be any value.
+ *
+ * @alias GenericParameter
+ * @constructor
+ * @extends FunctionParameter
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Terria} options.terria The Terria instance.
+ * @param {String} options.id The unique ID of this parameter.
+ * @param {String} [options.name] The name of this parameter.  If not specified, the ID is used as the name.
+ * @param {String} [options.description] The description of the parameter.
+ * @param {Boolean} [options.defaultValue] The default value.  If not specified, the first possible value is the default.
+ */
+var GenericParameter = function(options) {
```

Comment:
We have a `GenericParameterEditor` as a catch-all, but for the actual parameter types I think it would be better to be specific.  This isn't a generic parameter, it's a `TextParameter`, right?